contracts/epochs/FlowEpoch.cdc

```diff
@@ -509,14 +504,14 @@ pub contract FlowEpoch {
             proposedMetadata.setTotalRewards(proposedPayout)
             self.saveEpochMetadata(proposedMetadata)
         }
+        return rewardsSummary
     }
 
     /// Pays rewards to the nodes and delegators of the previous epoch
-    access(account) fun payRewards(_ totalRewards: UFix64) {
+    access(account) fun payRewards(_ rewardsSummary: FlowIDTableStaking.EpochRewardsSummary) {
         if let previousEpochMetadata = self.getEpochMetadata(self.currentEpochCounter - (1 as UInt64)) {
             if !previousEpochMetadata.rewardsPaid {
-                let rewardsBreakdownArray = previousEpochMetadata.rewardAmounts
-                self.borrowStakingAdmin().payRewards(rewardsBreakdownArray, totalRewards: totalRewards)
+                self.borrowStakingAdmin().payRewards(rewardsSummary)
```

Comment:
> @jordanschalm I just pushed some new changes, mostly making that new struct that also includes the total rewards amount: [4d78105](https://github.com/onflow/flow-core-contracts/pull/303/commits/4d781056f74d9947d6ad7446c4ca70bff6eedead)
> It is still a little bit awkward, but let me know if you think it is better.

Looks good, for me at least this structure is easier to follow 🙏.

My only other concern is that here, in `payRewards` the `EpochRewardsSummary` is passed in as an argument and this function assumes that it is for the previous epoch. It would be great to avoid the implicit coupling between the reward summary and epoch in the scope of `payRewards`.

Suggestion 1: Is it possible at this point to read both `rewardAmounts` and `totalRewards` from `previousEpochMetadata` (like we do in the [admin transaction](https://github.com/onflow/flow-core-contracts/pull/303/commits/4d781056f74d9947d6ad7446c4ca70bff6eedead#diff-dc2590b4ab366fdfdd96171ab0c059cb6c6e008ed5f9215f7ebe6d46566ec9d5R11))
```
if !previousEpochMetadata.rewardsPaid {
  let rewardsSummary = FlowIDTableStaking.EpochRewardsSummary(totalRewards: previousEpochMetadata.totalRewards, breakdown: previousEpochMetadata.rewardAmounts)
  self.borrowStakingAdmin().payRewards(rewardsSummary)
}
```
* We could fully remove the argument to `payRewards` and rename it `payRewardsForPreviousEpoch`

Suggestion 2: Alternatively, we could include an epoch counter in `EpochRewardsSummary` so that `payRewards` can sanity check that the `previousEpochMetadata` read from storage is consistent with the input argument. I don't think this is simple to do though, since `EpochRewardsSummary` needs to be instantiated in `FlowIDTableStaking` as well, which doesn't know about the epoch counter.

Reply:
I think I like what you're saying in Suggestion 1. It seemed a bit awkward to me also. I changed it in my latest commit to do that